app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -269,11 +271,37 @@ def build_metric_id(type, resource, metric_id)
     end
 
     def self.update_alert(*args)
-      _log.debug("Updating an alert on Hawkular provider: #{args}")
+      operation = args[0][:operation]
+      alert = args[0][:alert]
+      miq_alert = {
+        :id          => alert[:id],
+        :enabled     => alert[:enabled],
+        :description => alert[:description],
+        :conditions  => alert[:expression]
+      }
+      MiddlewareManager.find_each { |m| m.alert_manager.process_alert(operation, miq_alert) }
     end
 
     def self.update_alert_profile(*args)
-      _log.debug("Updating an alert profile on Hawkular provider: #{args}")
+      alert_profile_arg = args[0]
+      miq_alert_profile = {
+        :id                  => alert_profile_arg[:profile_id],
+        :old_alerts_ids      => alert_profile_arg[:old_alerts],
+        :new_alerts_ids      => alert_profile_arg[:new_alerts],
+        :old_assignments_ids => process_old_assignments_ids(alert_profile_arg[:old_assignments]),
+        :new_assignments_ids => process_new_assignments_ids(alert_profile_arg[:new_assignments])
+      }
+      MiddlewareManager.find_each do |m|
+        m.alert_profile_manager.process_alert_profile(alert_profile_arg[:operation], miq_alert_profile)
```

Comment:
self.update_alert() is invoked from the controller via an async task.
This is class method, as it should iterate over the MW providers to create/update the group trigger which is sync-ed.
Same for update_alert_profile, this should be operated over all the providers as it might use the "enterprise" wildcard.
There is a possible optimization for the cases where the middleware servers are specified that could search for the specific providers.
I didn't include it yet in this first iteration, I am focus first to provide a preliminar end-to-end flow and then improve the design.